File: nodejs/benchmarks-pagedjs.js
import { Bench } from "tinybench";
import { PlaywrightPdfCreator } from "./pdf-creator/playwright-pdf-creator.js";
import { PuppeteerPdfCreator } from "./pdf-creator/puppeteer-pdf-creator.js";
import { runBench, getContent, getPagedContent } from "./benchmark-utils.js";

export async function runBenchmarksPagedjs() {
  const playwrightPdfCreator = new PlaywrightPdfCreator();
  await playwrightPdfCreator.initialize();
  const puppeteerPdfCreator = new PuppeteerPdfCreator("chrome-headless-shell");
  await puppeteerPdfCreator.initialize();

  const content = await getContent();
  const pagedContent = await getPagedContent();
  const bench = new Bench();
  bench
    .add("PlaywrightNewPage", async () => {
      await playwrightPdfCreator.fromHtmlNewPage(content);
    })
    .add("PagedPlaywrightNewPage", async () => {
      await playwrightPdfCreator.fromHtmlNewPage(pagedContent);
    })
    .add("Playwright", async () => {
      await playwrightPdfCreator.fromHtml(content);
    })
    .add("PagedPlaywright", async () => {
      await playwrightPdfCreator.fromHtml(pagedContent);
    })
    .add("PuppeteerNewPage", async () => {
      await puppeteerPdfCreator.fromHtmlNewPage(content);
    })
    .add("PagedPuppeteerNewPage", async () => {
      await puppeteerPdfCreator.fromHtmlNewPage(pagedContent);
    })
    .add("Puppeteer", async () => {
      await puppeteerPdfCreator.fromHtml(content);
    })
    .add("PagedPuppeteer", async () => {
      await puppeteerPdfCreator.fromHtml(pagedContent);
    });
  await runBench(bench);

  await playwrightPdfCreator.destroy();
  await puppeteerPdfCreator.destroy();
}

File: c#/BenchmarksPagedjs.cs
using BenchmarkDotNet.Attributes;
using PdfBenchmarks.PdfCreator;

namespace PdfBenchmarks
{
    [MemoryDiagnoser]
    public class BenchmarksPagedjs
    {
        public enum ContentValue
        {
            Normal = 0,
            Paged = 1
        }

        [ParamsAllValues]
        public ContentValue ContentIndex { get; set; }

        private static List<string> ContentValues => [Data.GetContent(), Data.GetPagedContent()];

        private readonly PlaywrightPdfCreator _playwrightPdfCreator = new();
        private readonly PuppeteerSharpPdfCreator _puppeteerPdfCreator = new("chrome-headless-shell");

        [GlobalSetup]
        public async Task GlobalSetup()
        {
            await _playwrightPdfCreator.Initialize();
            await _puppeteerPdfCreator.Initialize();
        }

        [Benchmark]
        public async Task PlaywrightNewPage()
        {
            await _playwrightPdfCreator.FromHtmlNewPage(ContentValues[(int)ContentIndex]);
        }

        [Benchmark]
        public async Task Playwright()
        {
            await _playwrightPdfCreator.FromHtml(ContentValues[(int)ContentIndex]);
        }

        [Benchmark]
        public async Task PuppeteerNewPage()
        {
            await _puppeteerPdfCreator.FromHtmlNewPage(ContentValues[(int)ContentIndex]);
        }

        [Benchmark]
        public async Task Puppeteer()
        {
            await _puppeteerPdfCreator.FromHtml(ContentValues[(int)ContentIndex]);
        }
    }
}